File: 2540-minimum-common-value/2540-minimum-common-value.py
class Solution:
    def getCommon(self, nums1: List[int], nums2: List[int]) -> int:
        for n in nums1:
            l = 0
            r = len(nums2)-1
            while l <= r:
                m = (l+r) // 2
                if nums2[m] == n:
                    return n
                elif nums2[m] < n:
                    l = m + 1
                else:
                    r = m - 1
        return -1

File: 2540-minimum-common-value/2540-minimum-common-value.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number}
 */
var getCommon = function(nums1, nums2) {
    nums2 = new Set(nums2)
    for (const n of nums1) {
        if (nums2.has(n)) return n
    }
    return -1
};